pkg/api/getambassador.io/v2/handwritten.conversion.go

```diff
@@ -658,3 +659,53 @@ func Convert_v3alpha1_TCPMappingSpec_To_v2_TCPMappingSpec(in *v3alpha1.TCPMappin
 
 	return nil
 }
+
+func Convert_v2_TracingServiceSpec_To_v3alpha1_TracingServiceSpec(in *TracingServiceSpec, out *v3alpha1.TracingServiceSpec, s conversion.Scope) error {
+	if err := autoConvert_v2_TracingServiceSpec_To_v3alpha1_TracingServiceSpec(in, out, s); err != nil {
+		return err
+	}
+	// WARNING: in.TagHeaders requires manual conversion: does not exist in peer-type
+	// if only tag_headers are set, translate to custom_tags.
+	// if both are set, ignore tag_headers.
+	if in.TagHeaders != nil && in.V3CustomTags == nil {
+		out.CustomTags = []v3alpha1.TracingCustomTag{}
+		for _, tag := range in.TagHeaders {
+			out.CustomTags = append(out.CustomTags, v3alpha1.TracingCustomTag{
+				Tag: tag,
+				Header: &v3alpha1.TracingCustomTagTypeRequestHeader{
+					Name: tag,
+				},
+			})
+		}
+	}
+	return nil
+}
+
+func Convert_v3alpha1_TracingServiceSpec_To_v2_TracingServiceSpec(in *v3alpha1.TracingServiceSpec, out *TracingServiceSpec, s conversion.Scope) error {
+	in = in.DeepCopy()
+	// if only tag_headers are set, translate to custom_tags.
+	// if both are set, log a warning and ignore tag_headers.
+	if in.DeprecatedTagHeaders != nil {
+		if in.CustomTags == nil {
+			in.CustomTags = []v3alpha1.TracingCustomTag{}
+			for _, tag := range in.DeprecatedTagHeaders {
+				in.CustomTags = append(in.CustomTags, v3alpha1.TracingCustomTag{
+					Tag: tag,
+					Header: &v3alpha1.TracingCustomTagTypeRequestHeader{
+						Name: tag,
+					},
+				})
+			}
+		} else {
+			// TODO: Use dlog logger
```

Comment:
Let's use `dlog` here and then just use `context.Background` since this function doesn't have access to any other context. This way we are consistent with the rest of the code base and it gets outputted in a consistent format.